src/main/java/net/snowflake/client/core/arrow/fullvectorconverters/ArrowFullVectorConverter.java

```diff
@@ -0,0 +1,125 @@
+            int sfScale = Integer.parseInt(scaleStr);
+            if (sfScale != 0) {
+              return Types.MinorType.DECIMAL;
+            }
+            break;
+          }
+        case TIME:
+          return Types.MinorType.TIMEMILLI;
+        case TIMESTAMP_LTZ:
+          {
+            String scaleStr = vector.getField().getMetadata().get("scale");
+            int sfScale = Integer.parseInt(scaleStr);
+            switch (sfScale) {
+              case 0:
+                return Types.MinorType.TIMESTAMPSECTZ;
+              case 3:
+                return Types.MinorType.TIMESTAMPMILLITZ;
+              case 6:
+                return Types.MinorType.TIMESTAMPMICROTZ;
+              case 9:
+                return Types.MinorType.TIMESTAMPNANOTZ;
+            }
+            break;
+          }
+        case TIMESTAMP_TZ:
+          {
+            String scaleStr = vector.getField().getMetadata().get("scale");
+            int sfScale = Integer.parseInt(scaleStr);
+            switch (sfScale) {
+              case 0:
+                return Types.MinorType.TIMESTAMPSECTZ;
+              case 3:
+                return Types.MinorType.TIMESTAMPMILLITZ;
+              case 6:
+                return Types.MinorType.TIMESTAMPMICROTZ;
+              case 9:
+                return Types.MinorType.TIMESTAMPNANOTZ;
+            }
+            break;
+          }
+        case TIMESTAMP_NTZ:
+          {
+            String scaleStr = vector.getField().getMetadata().get("scale");
+            int sfScale = Integer.parseInt(scaleStr);
+            switch (sfScale) {
+              case 0:
+                return Types.MinorType.TIMESTAMPSEC;
+              case 3:
+                return Types.MinorType.TIMESTAMPMILLI;
+              case 6:
+                return Types.MinorType.TIMESTAMPMICRO;
+              case 9:
+                return Types.MinorType.TIMESTAMPNANO;
+            }
+            break;
+          }
+      }
+    }
+    return type;
+  }
+
+  static FieldVector convert(
+      RootAllocator allocator,
+      ValueVector vector,
+      DataConversionContext context,
+      SFBaseSession session,
+      int idx,
+      Object targetType)
+      throws SnowflakeSQLException {
+    try {
+      if (targetType == null) {
+        targetType = deduceType(vector);
+      }
+      if (targetType instanceof Types.MinorType) {
+        switch ((Types.MinorType) targetType) {
+          case TINYINT:
+            return new TinyIntVectorConverter(allocator, vector, context, session, idx).convert();
+          case SMALLINT:
+            return new SmallIntVectorConverter(allocator, vector, context, session, idx).convert();
+          case INT:
+            return new IntVectorConverter(allocator, vector, context, session, idx).convert();
+          case BIGINT:
+            return new BigIntVectorConverter(allocator, vector, context, session, idx).convert();
+          case DECIMAL:
```

Comment:
Should we set default for unrecognised types?

Reply:
There is no reasonable default beyond null or an exception, so I think at most we could discuss which of these two is better

Reply:
My thought is about exception of unsupported type